### ISSUE TITLE
(<Android 10) Suggestion to add reference for Compatibility and notes on SQLite Version and Android Version Matrix

### ISSUE THREAD

**Issue by PistonShot1** [tekartik/sqflite#1133]
After doing testing for android 10 , is when I realised there a certain matters needed to be considered, when using SQLite, 

For instance, UPSERT statement ( INSERT INTO ..... ON  CONFLICT UPDATE SET) will not work for android 10 (API Level 29) as it will be using SQLite version 3.22, and 3.22 doesnt have support yet for UPSERT, only on 3.24 and above , [reference](https://www.sqlite.org/lang_upsert.html)

Android Version and SQLite Version Matrix : 
https://developer.android.com/reference/android/database/sqlite/package-summary

Please add a warning or info alert , to the pub.dev readme , this would help a lot for new developers that are considering to use sqlite

UPSERT is just one of the feature, there are many other difference, unfortunately many developer would miss and not look at the specifics , adding the warning might give awareness at least before starting to implement

**Comment by alextekartik** [comment]
Thanks for the note. Indeed sqflite uses the SQLite available on the platform so some features might not be available. I have added some notes in the [version page](https://github.com/tekartik/sqflite/blob/master/sqflite/doc/version.md) and link to this page from the main README.md`

**Comment by PistonShot1** [comment]
@alextekartik , I would like to verify a few matter, so it is possible to override the existing version sqlite bundled by native android and iOS to use a newer version instead? Based on you suggestion , using [sqflite_common_ffi](https://pub.dev/packages/sqflite_common_ffi) or [sqlite3](https://pub.dev/packages/sqlite3) , is that correct ?

**Comment by alextekartik** [comment]
@PistonShot1 That's correct see https://github.com/tekartik/sqflite/blob/master/sqflite_common_ffi/doc/using_ffi_instead_of_sqflite.md#using-sqflite_ffi-instead-of-sqflite

You should add sqflite_common_ffi (that uses sqlite3) and sqlite3_flutter_libs

Plesase, prefer opening a new issue instead of re-opening a closed issue (somehow related but different).

**Comment by PistonShot1** [comment]
Thank you for your prompt reply.
